pcsx2/Frontend/InputManager.cpp

```diff
@@ -669,11 +669,24 @@ bool InputManager::InvokeEvents(InputBindingKey key, float value, GenericInputBi
 				continue;
 
 			const u8 bit = static_cast<u8>(1) << i;
-			const bool negative = binding->keys[i].negative;
+			const bool negative = binding->keys[i].modifier == InputModifier::Negate;
 			const bool new_state = (negative ? (value < 0.0f) : (value > 0.0f));
 
-			// invert if we're negative, since the handler expects 0..1
-			const float value_to_pass = (negative ? ((value < 0.0f) ? -value : 0.0f) : (value > 0.0f) ? value : 0.0f);
+			float value_to_pass = 0.0f;
+			switch (binding->keys[i].modifier)
+			{
+				case InputModifier::None:
+					if (value > 0.0f)
+						value_to_pass = value;
+					break;
+				case InputModifier::Negate:
+					if (value < 0.0f)
+						value_to_pass = -value;
+					break;
+				case InputModifier::FullAxis:
+					value_to_pass = value * 0.5f + 0.5f;
```

Comment:
I think what you want to do here instead is add 32768 and then divide by two?
Example input: -32768 (button released)
new = (-32768 + 32768)/2

Reply:
Inputs are [normalized fairly early on](https://github.com/tellowkrinkle/pcsx2/blob/b410447bdab63ed0b54a7745cd420c7ccd939cb6/pcsx2/Frontend/SDLInputSource.cpp#L514) so at that point they're between -1 and 1

Not perfect, since that means we scale values less than 0 slightly differently from values greater than 0 (rather than adjusting the original number to 0-65535 before normalizing) but it should be close enough.